docs/guides/register_packages_on_chain.md

```diff
@@ -1,39 +1,84 @@
 The Autonolas ecosystem has an **on-chain protocol** that provides primitives to create, operate and secure agent services on a blockchain. It also provides a mechanism that incentivizes their creation and rewards developers and service operators proportionally for their efforts to support the growth of the Autonolas ecosystem.
 
 The **on-chain registry** is a collection of ERC721 smart contracts (eventually deployed on all major blockchains) that handle the life cycle of **packages** (that is, agent components, agents and agent services) within the on-chain protocol. These contracts store a representation of the code developed in the form of an NFT.
+Therefore, registering and securing a package in the on-chain registry is the required step so that they can be used in the on-chain protocol.
 
-Therefore, registering a package in the on-chain registry is the required step so that they can be used in the on-chain protocol.
-
-The on-chain protocol can be accessed through the {{on_chain_frontend}}.
+The on-chain registry can be accessed through the {{on_chain_frontend}}.
 
 ## What will you learn
 In this guide, you will learn how to:
 
-  * Register agent components, agents and agent services in the on-chain registry.
+  * Register packages (agent components, agents and agent services) in the on-chain registry.
   * Manage the life cycle of an agent service on-chain.
 
 ## Requirements
 This guide is based in the on-chain registry currently deployed in the [Görli testnet](https://goerli.net/). Ensure that you have:
 
   * A **[Görli testnet](https://goerli.net/) wallet address** (e.g., [Metamask](https://metamask.io/)) with some GörliETH funds in it. You can use, for example, a [Görli POW faucet](https://goerli-faucet.pk910.de/) to earn some free GörliETH.
-  * The remote registry **package hash** of the already published completed package that you want to register on-chain. See the corresponding guides in case you are in doubt on how to create and publish a package in a remote registry.
+  * The **package hash** of the agent component, agent or agent service that you want to register on-chain, and which should be already published on a remote registry. See the corresponding guides in case you are in doubt on how to create and publish a package in a remote registry.
   * An **NFT image URL**. This image will be used to represent the NFT on marketplaces such as [OpenSea](https://opensea.io/). You can use [this sample image URL](https://gateway.autonolas.tech/ipfs/Qmbh9SQLbNRawh9Km3PMEDSxo77k1wib8fYZUdZkhPBiev) for testing purposes.
   * If you are registering an agent service, you will also need the [Görli testnet](https://goerli.net/) **addresses of the agents** that are part of the service.
 
 
+## How packages are registered using the protocol frontend
+There are two objects published in the remote registry that need to be considered when registering packages in the on-chain registry:
+
```

Comment:
There is some inconsistency: at some places we refer to "the registry" and in other places to "a registry". Maybe to avoid confusion we should sticj to only one of those depending on what we want to convey.